src/libcrun/container.c

```diff
@@ -2508,7 +2508,7 @@ libcrun_container_run_internal (libcrun_container_t *container, libcrun_context_
         return ret;
     }
 
-  if (context->console_socket)
+  if (context->console_socket && def->process && def->process->terminal)
```

Comment:
could we do `if (context->console_socket >= 0)` here so that if the conditions for creating the console socket are changed we don't need to keep them in sync in two different places?

Reply:
context->console_socket is a string obtained from the environment variable, not a socket fd. context->console_socket>=0 will cause a direct crash.
if (context->console_socket && def->process && def->process->terminal) is used to determine whether the console socket client needs to be initialized in the main process.